tests/unit/test_dimension_utilities.py

```diff
@@ -679,6 +679,38 @@ def test_is_index_subset(self):
                 )
             )
 
+        with self.subTest('No index Subset'):
```

Comment:
Nitpick: I think it's probably worth explicitly stating this subtest is for an empty `dimensions` rather than "No index subset".